### ISSUE TITLE
Duplicate parameters in Fn.Def

### ISSUE THREAD

**Issue by mougino** [RFO-BASIC/Basic#200]
In BASIC! V01.88 (and before) the user is allowed to write that:
```VB
FN.DEF myFunc(a,b,a)
  PRINT a
  PRINT b
  PRINT a
FN.END

CALL myFunc(1,2,3)
```
In any other language this would throw an error, I think this should not be tolerated in BASIC! as well (mostly could be due to a user typo and worth reporting to him).

My proposal to fix this requires a HashSet and takes place at the beginning of `private boolean  executeFN_DEF()`:
```Java
		String var = getNewFNVar();									// Get the function name
		if (var == null)				return false;
		int fVarNumber = createNewVar(var);							// Save the VarNumber of the function name
		VarType fType = VarIsNumeric ? VarType.NUM : VarType.STR;

		ArrayList<FunctionParameter> fParms = new ArrayList<FunctionParameter>();
		Set<String> uparms = new HashSet<String>();    // List of unique parameter names
		if (!isNext(')')) {
			do {													// Get each of the parameter names
				String name = parseVar(!USER_FN_OK);				// without creating any new vars
				if (name == null)		return false;
				if (VarIsArray && !isNext(']')) {					// Process array
					return RunTimeError(EXPECT_ARRAY_NO_INDEX);		// Array must not have any indices
				}
				uparms.add(name);
				VarType type = VarIsNumeric ? VarType.NUM : VarType.STR;
				fParms.add(new FunctionParameter(name, type, VarIsArray));
			} while (isNext(','));
			if ( !(isNext(')') && checkEOL()) ) return false;
		}
		if (uparms.size() < fParms.size()) { return RunTimeError("Duplicate parameter names at:"); }

// [...]
```
You need to add an `import java.util.HashSet;` at the beginning of `Run.java` as well (next to the `import java.util.Set;` for example)

This also will prevent creating potential variables out of their name space with issue #198 fix proposal.

**Comment by jMarcS** [comment]
> In any other language this would throw an error, I think this should not be tolerated in BASIC! as well 

I agree. In v01.88 and before, the effect is to create a second variable with the same name. The linear search can never find the second one, so the duplicate parameter is ignored. The value passed to initialize the second parameter is useless.

> This also will prevent creating potential variables out of their name space with issue #198 fix proposal.

With that proposed change, this bug goes from merely inconvenient to catastrophic. At best, the calling function's data gets corrupted. At worst, the `VarNames` absolute index becomes negative; I suspect that would crash BASIC!. See the full discussion in issue #198.

In short, this bug MUST be fixed. Nicolas has provided an easy way to fix it.

**Comment by jMarcS** [comment]
Released in v01.89, 2015/12/29.
